src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -206,6 +208,29 @@ private boolean areAllChildrenIgnored() {
     }
 
     /**
+     * Returns a {@link Statement}: notify all non-overridden {@code @Ignore} methods on this class
+     * and superclasses before executing {@code next}.
+     * @param next The next statement, or {@code null}, if there is no next statement.
+     * @param notifier The runner notifier.
+     */
+    protected Statement withIgnores(final Statement next, final RunNotifier notifier) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                for (final T each : getFilteredChildren()) {
```

Comment:
I can imagine cases where users of JUnit expect the notifier to be called from the scheduler (say, if the scheduler sets a thread local variable).

Could we just change `runChildren` to take in an `Iterable<T>` and use it here?